### PR TITLE
8290990: Clear .root style class from a root node that is removed from a Scene/SubScene

### PR DESCRIPTION
**Pull request by mstr2**
When a node is set as the root node of a `Scene` or `SubScene`, the "root" style class is automatically added to the node, but not cleared when the node is later removed from the scene.

This can lead to an incorrectly set "root" style class when the removed node is inserted in the scene graph below the root node, or to duplicate "root" style class entries when the removed node is again set as the root node of the scene.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8290990](https://bugs.openjdk.org/browse/JDK-8290990): Clear .root style class from a root node that is removed from a Scene/SubScene

### Reviewers
 * [John Hendrikx](https://openjdk.org/census#jhendrikx) (@hjohn - Author)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)
 * [Marius Hanl](https://openjdk.org/census#mhanl) (@Maran23 - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/849/head:pull/849` \
`$ git checkout pull/849`

Update a local copy of the PR: \
`$ git checkout pull/849` \
`$ git pull https://git.openjdk.org/jfx pull/849/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 849`

View PR using the GUI difftool: \
`$ git pr show -t 849`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/849.diff">https://git.openjdk.org/jfx/pull/849.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
This seems like a pretty safe and obvious fix. Since it touches CSS, it needs a second reviewer, but I don't expect it will take long to review.

/reviewers 2

**Review by hjohn — APPROVED** [review]
I've ran into a problem related to this before (multiple root's), and this looks like a good fix.

**Review by aghaisas — APPROVED** [review]
Looks good to me.

File: modules/javafx.graphics/src/main/java/javafx/scene/SubScene.java [hunk 1]
```diff
@@ -316,6 +316,7 @@ protected void invalidated() {
                     if (oldRoot != null) {
                         StyleManager.getInstance().forget(SubScene.this);
```
Comment (Maran23) [inline modules/javafx.graphics/src/main/java/javafx/scene/SubScene.java:317]: Not part of the story, but interesting that here the sub scene will be 'forgotten' but nothing like this happens in the `Scene` class. 😄

**Review by Maran23 — APPROVED** [review]
Looks good to me too.

**Pull request closed without merging**